pkg/pillar/cmd/volumemgr/handlediskmetrics.go

```diff
@@ -0,0 +1,266 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package volumemgr
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/diskmetrics"
+	"github.com/lf-edge/eve/pkg/pillar/flextimer"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/utils"
+	"github.com/shirou/gopsutil/disk"
+)
+
+func publishDiskMetrics(ctx *volumemgrContext, statuses ...*types.DiskMetric) {
+	for _, status := range statuses {
+		key := status.Key()
+		log.Debugf("publishDiskMetrics(%s)", key)
+		pub := ctx.pubDiskMetric
+		pub.Publish(key, *status)
+		log.Debugf("publishDiskMetrics(%s) Done", key)
+	}
+}
+
+func unpublishDiskMetrics(ctx *volumemgrContext, statuses ...*types.DiskMetric) {
+	for _, status := range statuses {
+		key := status.Key()
+		log.Debugf("unpublishDiskMetrics(%s)", key)
+		pub := ctx.pubDiskMetric
+		c, _ := pub.Get(key)
+		if c == nil {
+			log.Errorf("unpublishDiskMetrics(%s) not found", key)
+			continue
+		}
+		pub.Unpublish(key)
+		log.Debugf("unpublishDiskMetrics(%s) Done", key)
+	}
+}
+
+func lookupDiskMetric(ctx *volumemgrContext, key string) *types.DiskMetric {
+	key = pathToKey(key)
+	log.Debugf("lookupDiskMetric(%s)", key)
+	pub := ctx.pubDiskMetric
+	c, _ := pub.Get(key)
+	if c == nil {
+		log.Debugf("lookupDiskMetric(%s) not found", key)
+		return nil
+	}
+	status := c.(types.DiskMetric)
+	log.Debugf("lookupDiskMetric(%s) Done", key)
+	return &status
+}
+
+func publishAppDiskMetrics(ctx *volumemgrContext, statuses ...*types.AppDiskMetric) {
+	for _, status := range statuses {
+		key := status.Key()
+		log.Debugf("publishAppDiskMetrics(%s)", key)
+		pub := ctx.pubAppDiskMetric
+		pub.Publish(key, *status)
+		log.Debugf("publishAppDiskMetrics(%s) Done", key)
+	}
+}
+
+func unpublishAppDiskMetrics(ctx *volumemgrContext, statuses ...*types.AppDiskMetric) {
+	for _, status := range statuses {
+		key := status.Key()
+		log.Debugf("unpublishAppDiskMetrics(%s)", key)
+		pub := ctx.pubAppDiskMetric
+		c, _ := pub.Get(key)
+		if c == nil {
+			log.Errorf("unpublishAppDiskMetrics(%s) not found", key)
+			continue
+		}
+		pub.Unpublish(key)
+		log.Debugf("unpublishAppDiskMetrics(%s) Done", key)
+	}
+}
+
+func lookupAppDiskMetric(ctx *volumemgrContext, key string) *types.AppDiskMetric {
+	key = pathToKey(key)
+	log.Debugf("lookupAppDiskMetric(%s)", key)
+	pub := ctx.pubAppDiskMetric
+	c, _ := pub.Get(key)
+	if c == nil {
+		log.Debugf("lookupAppDiskMetric(%s) not found", key)
+		return nil
+	}
+	status := c.(types.AppDiskMetric)
+	log.Debugf("lookupAppDiskMetric(%s) Done", key)
+	return &status
+}
+
+//diskMetricsTimerTask calculates and publishes disk metrics periodically
+func diskMetricsTimerTask(ctx *volumemgrContext) {
+	log.Infoln("starting report diskMetricsTimerTask timer task")
+	createOrUpdateDiskMetrics(ctx)
+
+	diskMetricInterval := time.Duration(ctx.globalConfig.GlobalValueInt(types.DiskScanMetricInterval)) * time.Second
+	max := float64(diskMetricInterval)
```

Comment:
This doesn't seem to react to changes to the configuration.
In handlemetrics.go we have logic to do that for  updateMetricsTimer plus code in parseConfigItem to call it.
That makes sense for this timer as well.

Reply:
Changed.